internal/controller/component_customizer.go

```diff
@@ -110,6 +110,10 @@ func customizeDeployment(pSpec operatorv1.ProviderSpec, d *appsv1.Deployment) {
 			d.Spec.Template.Spec.Tolerations = dSpec.Tolerations
 		}
 
+		if dSpec.ImagePullSecrets != nil {
```

Comment:
Probably the best way to deal with the lint issue is to add `//nolint:nestif` before the if block on line 94